server/src/userManagement.mjs

```diff
@@ -117,6 +119,59 @@ function heartbeat(ws) {
 }
 
 function addUser(userId) {
+  userId = "" + userId;
+  var users = getUsers();
+  let appId,user,group;
+
+  //getting user, group and appId from userId
+  if (userId.includes("~")){
```

Comment:
Same comment as above in terms of seemingly duplicative userid parsing code.

Reply:
Added a utility function that returns an object containing the split userId as user, group, and/or appId